src/main/java/com/rcv/GuiConfigController.java

```diff
@@ -251,12 +279,40 @@ private void saveElectionConfig(File saveFile) {
         choiceMultiSeatTransferRule.getValue() != null
             ? choiceMultiSeatTransferRule.getValue().toString()
             : Tabulator.MultiSeatTransferRule.TRANSFER_RULE_UNKNOWN.toString();
-    rules.maxRankingsAllowed = Integer.parseInt(textFieldMaxRankingsAllowed.getText());
-    rules.maxSkippedRanksAllowed = Integer.parseInt(textFieldMaxSkippedRanksAllowed.getText());
-    rules.numberOfWinners = Integer.parseInt(textFieldNumberOfWinners.getText());
-    rules.decimalPlacesForVoteArithmetic = Integer
-        .parseInt(textFieldDecimalPlacesForVoteArithmetic.getText());
-    rules.minimumVoteThreshold = Integer.parseInt(textFieldMinimumVoteThreshold.getText());
+    rules.maxRankingsAllowed =
+        !textFieldMaxRankingsAllowed.getText().isEmpty()
+            ? Integer.parseInt(textFieldMaxRankingsAllowed.getText())
+            : null;
+    rules.maxSkippedRanksAllowed =
+        !textFieldMaxSkippedRanksAllowed.getText().isEmpty()
+            ? Integer.parseInt(textFieldMaxSkippedRanksAllowed.getText())
+            : null;
+    rules.numberOfWinners =
+        !textFieldNumberOfWinners.getText().isEmpty()
+            ? Integer.parseInt(textFieldNumberOfWinners.getText())
+            : ElectionConfig.DEFAULT_NUMBER_OF_WINNERS;
+    rules.decimalPlacesForVoteArithmetic =
+        !textFieldDecimalPlacesForVoteArithmetic.getText().isEmpty()
+            ? Integer.parseInt(textFieldDecimalPlacesForVoteArithmetic.getText())
+            : ElectionConfig.DEFAULT_DECIMAL_PLACES_FOR_VOTE_ARITHMETIC;
+    rules.minimumVoteThreshold =
+        !textFieldMinimumVoteThreshold.getText().isEmpty()
+            ? Integer.parseInt(textFieldMinimumVoteThreshold.getText())
+            : ElectionConfig.DEFAULT_MINIMUM_VOTE_THRESHOLD.intValue();
+    rules.batchElimination =
+        ((RadioButton) toggleBatchElimination.getSelectedToggle()).getText().equals("True");
```

Comment:
Could also factor out this bit of repeated logic.

Reply:
Done in latest commit, along with additional refactoring. Please re-approve when you get a chance to review. 💯